assets/js/blocks/product-category/edit-mode.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { Button, Placeholder } from '@wordpress/components';
+import { Icon, file } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import ProductCategoryControl from '@woocommerce/editor-components/product-category-control';
+
+/**
+ * Internal dependencies
+ */
+import { Props } from './types';
+
+export interface EditModeProps extends Props {
+	isEditing: boolean;
+	setIsEditing: ( isEditing: boolean ) => void;
+	changedAttributes: object;
+	setChangedAttributes: ( changedAttributes: object ) => void;
+}
+
+export const ProductsByCategoryEditMode = (
+	props: EditModeProps
+): JSX.Element => {
+	const {
+		debouncedSpeak,
+		setIsEditing,
+		changedAttributes,
+		setChangedAttributes,
+		attributes,
+	} = props;
+
+	const currentAttributes = { ...attributes, ...changedAttributes };
+
+	const stopEditing = () => {
+		setIsEditing( false );
+		setChangedAttributes( {} );
+	};
+
+	const save = () => {
+		const { setAttributes } = props;
+
+		setAttributes( changedAttributes );
+		stopEditing();
+	};
+
+	const onDone = () => {
+		save();
+		debouncedSpeak(
+			__(
+				'Showing Products by Category block preview.',
+				'woo-gutenberg-products-block'
+			)
+		);
+	};
+
+	const onCancel = () => {
+		stopEditing();
+		debouncedSpeak(
+			__(
+				'Showing Products by Category block preview.',
+				'woo-gutenberg-products-block'
+			)
+		);
+	};
+
+	return (
+		<Placeholder
+			icon={ <Icon icon={ file } /> }
+			label={ __(
+				'Products by Category',
+				'woo-gutenberg-products-block'
+			) }
+			className="wc-block-products-grid wc-block-products-category"
+		>
+			{ __(
+				'Display a grid of products from your selected categories.',
+				'woo-gutenberg-products-block'
+			) }
+			<div className="wc-block-products-category__selection">
+				<ProductCategoryControl
```

Comment:
![image](https://user-images.githubusercontent.com/4463174/179233010-81f4dbd5-b9c5-4e9e-96de-4a7ce8464f45.png)

To avoid introducing new TS errors, maybe we could add a JSDoc and define some props optional. What do you think?